src/pages/Auth/SignUp.tsx

```diff
@@ -1,112 +1,86 @@
-import { ChangeEvent, useState } from "react";
+import { useEffect, useState } from "react";
+import StepWizard from "react-step-wizard";
 
-import { signUp } from "../../api";
+import { UserData, UserPlan, Workspace } from "./Wizard";
+
+import transitionsStyles from "./Wizard/transitions.module.scss";
+import { useRegisterMutation } from "@api";
+import { AUTH_TOKEN, REFRESH_TOKEN, useLocalStorage } from "@hooks";
 
 export const SignUp = () => {
-  const [userForm, setUserForm] = useState({
-    name: "",
-    email: "",
-    password: "",
-    role: "",
-  });
-  const [workspaceData, setWorkspaceData] = useState({
-    name: "",
-    description: "",
+  const [state, setState] = useState({
+    user: {
+      name: "",
+      email: "",
+      password: "",
+      surname: "",
+      secondName: "",
+    },
+    workspace: {
+      name: "",
+      description: "",
+    },
+    plan: "",
   });
+  const transitions = {
+    enterRight: `${transitionsStyles.animated} ${transitionsStyles.enterRight}`,
+    enterLeft: `${transitionsStyles.animated} ${transitionsStyles.enterLeft}`,
+    exitRight: `${transitionsStyles.animated} ${transitionsStyles.exitRight}`,
+    exitLeft: `${transitionsStyles.animated} ${transitionsStyles.exitLeft}`,
+    intro: `${transitionsStyles.animated} ${transitionsStyles.intro}`,
+  };
+  const [, setAuthToken] = useLocalStorage(AUTH_TOKEN);
+  const [, setRefresh] = useLocalStorage(REFRESH_TOKEN);
+  const [, setUser] = useLocalStorage("user");
+  const [register, { data, status }] = useRegisterMutation();
+
+  useEffect(() => {
+    if (data && status === "fulfilled") {
+      setAuthToken(data.authToken);
+      setRefresh(data.refreshToken);
+      setUser(data.user);
+    }
+  }, [data, setAuthToken, setRefresh, setUser, status]);
+
+  const updateForm = (stateKey: string, value: string, key?: string) => {
+    const newState = { ...state };
+
+    if (key) {
+      newState[stateKey][key] = value;
+    } else {
+      newState[stateKey] = value;
+    }
+    setState({
```

Comment:
If  You  need  previous  state  to  update  new one  -  don't use  directly  old state  like  this  .
You  need  to use  callback  setState(old=>{   return {...old, newKey:'value'}}).

Better  to split  this one  state  to  3  different